manager/controlapi/service.go

```diff
@@ -364,6 +395,7 @@ func (s *Server) CreateService(ctx context.Context, request *api.CreateServiceRe
 		return nil, err
 	}
 
+	// Check to see if the Secrets portion of the spec is valid
```

Comment:
Unrelated to this PR, but this should go into `validateServiceSpec`.

If you look at that function, it's already doing:
```go
 	if err := validateAnnotations(spec.Annotations); err != nil {
 		return err
 	}
 	if err := validateTask(spec.Task); err != nil {
 		return err
 	}
 	if err := validateUpdate(spec.Update); err != nil {
 		return err
 	}
 	if err := validateEndpointSpec(spec.Endpoint); err != nil {
 		return err
 	}
```

That function should call `validateSecretReferences` or something